flink-python/src/main/java/org/apache/flink/streaming/api/transformations/python/DelegateOperatorTransformation.java

```diff
@@ -0,0 +1,128 @@
+ * configurations include {@link OutputTag}s, {@code numPartitions} and general {@link
+ * Configuration}.
+ */
+public interface DelegateOperatorTransformation<OUT> {
+
+    SimpleOperatorFactory<OUT> getOperatorFactory();
+
+    static void configureDelegatedOperator(
+            DelegateOperatorTransformation<?> transformation,
+            AbstractPythonFunctionOperator<?> operator) {
+        DelegateOperator<?> delegateOperator =
+                (DelegateOperator<?>) transformation.getOperatorFactory().getOperator();
+
+        operator.getConfiguration().addAll(delegateOperator.getConfiguration());
+
+        if (operator instanceof DataStreamPythonFunctionOperator) {
+            DataStreamPythonFunctionOperator<?> dataStreamOperator =
+                    (DataStreamPythonFunctionOperator<?>) operator;
+            dataStreamOperator.addSideOutputTags(delegateOperator.getSideOutputTags());
+            if (delegateOperator.getNumPartitions() != null) {
+                dataStreamOperator.setNumPartitions(delegateOperator.getNumPartitions());
+            }
+        }
+    }
+
+    /**
+     * {@link DelegateOperator} holds configurations, e.g. {@link OutputTag}s, which will be applied
+     * to the actual python operator at translation stage.
+     */
+    class DelegateOperator<OUT> extends AbstractPythonFunctionOperator<OUT>
+            implements DataStreamPythonFunctionOperator<OUT> {
+
+        private final Map<String, OutputTag<?>> sideOutputTags = new HashMap<>();
+        private @Nullable Integer numPartitions = null;
+
+        public DelegateOperator() {
+            super(new Configuration());
+        }
+
+        @Override
+        public void addSideOutputTags(Collection<OutputTag<?>> outputTags) {
+            for (OutputTag<?> outputTag : outputTags) {
+                sideOutputTags.put(outputTag.getId(), outputTag);
+            }
+        }
+
+        @Override
+        public Collection<OutputTag<?>> getSideOutputTags() {
+            return sideOutputTags.values();
+        }
+
+        @Override
+        public void setNumPartitions(int numPartitions) {
+            this.numPartitions = numPartitions;
+        }
+
+        @Nullable
+        public Integer getNumPartitions() {
+            return numPartitions;
+        }
+
+        @Override
+        public TypeInformation<OUT> getProducedType() {
+            throw new RuntimeException("This should not be invoked on a DelegateOperator!");
+        }
+
+        @Override
+        public DataStreamPythonFunctionInfo getPythonFunctionInfo() {
+            throw new RuntimeException("This should not be invoked on a DelegateOperator!");
+        }
+
+        @Override
+        public <T> DataStreamPythonFunctionOperator<T> copy(
+                DataStreamPythonFunctionInfo pythonFunctionInfo,
+                TypeInformation<T> outputTypeInfo) {
+            throw new RuntimeException("This should not be invoked on a DelegateOperator!");
```

Comment:
May be we should implement this. This method is used when performing operator chain optimization.

Reply:
Currently python broadcast opreator does not support chaining with downstream, so I guess this check could be remove in another PR that supports broadcast chaining.